byway/src/main/webapp/routepage.js

```diff
@@ -186,24 +211,23 @@ function renderRouteList(){
 }
 
 /** Creates a button in the schedule panel in the html
- *  @param {String} waypoint a String to add as a button 
+ *  @param {PlaceResult} waypoint a PlaceResult object whose name property is added as a button 
  *  @return {button} routeBtn a button showing a selected waypoint along route
  */
 function createRouteButton(waypoint){
   const routeBtn = document.createElement('button');
-  routeBtn.innerText = waypoint;
-  if(!destinations.includes(waypoint)){
-    routeBtn.className =  "btn stop-btn";
-  } else{
+  routeBtn.innerText = waypoint.name;
+  if(destinations.some(destination => destination.name === waypoint.name)){
     routeBtn.className =  "btn destination-btn";
+  } else {
+    routeBtn.className =  "btn stop-btn";
   }
   routeBtn.addEventListener("click", function() {
     // only delete if the waypoint is only a stop, not a destination
-    if(!destinations.includes(waypoint)){
-      route = route.filter(stop => stop != waypoint);
+    if(!destinations.some(destination => destination.name === waypoint.name)){
```

Comment:
Can you get rid of this check and only add the event listener in the case where it is a stop? Its kind of redundant to have this check twice, and the event listener doesn't do anything for the destination buttons anyways.

Reply:
Thanks for catching that, it was redundant. Changed!